### DISCUSSION TITLE
Debug hangs at loading symbols

### DISCUSSION THREAD

**Discussion by jsjtczh** [microsoft/vscode-cpptools#7869] [Q&A]
When I try to debug a program, vs code seems hang at loading symbols of shared library. How can I solve it?

my launch.json: 
{
  // Use IntelliSense to learn about possible attributes.
  // Hover to view descriptions of existing attributes.
  // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
  "version": "0.2.0",
  "configurations": [
    {
      "name": "(lldb) Launch",
      "type": "cppdbg",
      "request": "launch",
      "program": "${workspaceFolder}/build/exe/reconstruction/main_automatic",
      "args": [],
      "stopAtEntry": false,
      "cwd": "${workspaceFolder}",
      "environment": [],
      "externalConsole": false,
      "MIMode": "lldb",
      "logging": {
        "engineLogging": true,
        "trace": true,
        "traceResponse": true
      }
    }
  ]
}

log:
--> E (output): {"type":"event","event":"output","body":{"category":"console","output":"1: (134) LaunchOptions{\"name\":\"(lldb) Launch\",\"type\":\"cppdbg\",\"request\":\"launch\",\"program\":\"/Users/czh190502/Projects/linyunjing/reconstruction/build/exe/reconstruction/main_automatic\",\"args\":[],\"stopAtEntry\":false,\"cwd\":\"/Users/czh190502/Projects/linyunjing/reconstruction\",\"environment\":[],\"externalConsole\":false,\"MIMode\":\"lldb\",\"logging\":{\"engineLogging\":true,\"trace\":true,\"traceResponse\":true},\"__configurationTarget\":5,\"__sessionId\":\"28aaae7c-657e-49ae-abd2-3d6dbc5113f8\",\"miDebuggerPath\":\"/Users/czh190502/.vscode/extensions/ms-vscode.cpptools-1.5.1/debugAdapters/lldb-mi/bin/lldb-mi\"}\n"},"seq":2}
1: (134) LaunchOptions{"name":"(lldb) Launch","type":"cppdbg","request":"launch","program":"/Users/czh190502/Projects/linyunjing/reconstruction/build/exe/reconstruction/main_automatic","args":[],"stopAtEntry":false,"cwd":"/Users/czh190502/Projects/linyunjing/reconstruction","environment":[],"externalConsole":false,"MIMode":"lldb","logging":{"engineLogging":true,"trace":true,"traceResponse":true},"__configurationTarget":5,"__sessionId":"28aaae7c-657e-49ae-abd2-3d6dbc5113f8","miDebuggerPath":"/Users/czh190502/.vscode/extensions/ms-vscode.cpptools-1.5.1/debugAdapters/lldb-mi/bin/lldb-mi"}
--> E (output): {"type":"event","event":"output","body":{"category":"console","output":"1: (220) Starting: \"/Users/czh190502/.vscode/extensions/ms-vscode.cpptools-1.5.1/debugAdapters/lldb-mi/bin/lldb-mi\" --interpreter=mi\n"},"seq":4}
1: (220) Starting: "/Users/czh190502/.vscode/extensions/ms-vscode.cpptools-1.5.1/debugAdapters/lldb-mi/bin/lldb-mi" --interpreter=mi
--> E (output): {"type":"event","event":"output","body":{"category":"console","output":"1: (242) DebuggerPid=18972\n"},"seq":6}
1: (242) DebuggerPid=18972
--> E (output): {"type":"event","event":"output","body":{"category":"console","output":"1: (355) ->(gdb)\n"},"seq":8}
1: (355) ->(gdb)
--> E (output): {"type":"event","event":"output","body":{"category":"console","output":"1: (378) <-1001-gdb-set auto-solib-add on\n"},"seq":10}
1: (378) <-1001-gdb-set auto-solib-add on
--> E (output): {"type":"event","event":"output","body":{"category":"console","output":"1: (378) ->1001^done\n"},"seq":12}
1: (378) ->1001^done
--> E (output): {"type":"event","event":"output","body":{"category":"console","output":"1: (379) ->(gdb)\n"},"seq":14}
1: (379) ->(gdb)
--> E (output): {"type":"event","event":"output","body":{"category":"console","output":"1: (382) 1001: elapsed time 5\n"},"seq":16}
1: (382) 1001: elapsed time 5
--> E (output): {"type":"event","event":"output","body":{"category":"console","output":"1: (383) <-1002-gdb-set solib-search-path \"/Users/czh190502/Projects/linyunjing/reconstruction/build/exe/reconstruction:\"\n"},"seq":18}
1: (383) <-1002-gdb-set solib-search-path "/Users/czh190502/Projects/linyunjing/reconstruction/build/exe/reconstruction:"
--> E (output): {"type":"event","event":"output","body":{"category":"console","output":"1: (384) ->1002^done\n"},"seq":20}
1: (384) ->1002^done
--> E (output): {"type":"event","event":"output","body":{"category":"console","output":"1: (384) 1002: elapsed time 0\n"},"seq":22}
1: (384) 1002: elapsed time 0
--> E (output): {"type":"event","event":"output","body":{"category":"console","output":"1: (384) ->(gdb)\n"},"seq":24}
1: (384) ->(gdb)

**Comment by djairao** [comment]
Are you running on Linux? If yes, are you able to run `gdb` from a shell prompt and debug this executable successfully?

**Reply by jsjtczh** [reply]
I'm running on Mac. I write an another simple project to test if it's my computer's problem,  and the debugger works pretty good. But in this large project, it cannot work.

**Reply by djairao** [reply]
Ok, then it must be something specific to your project. So that question is are you able to run `gdb` from the Mac terminal and debug this executable from the Mac terminal?

**Reply by jsjtczh** [reply]
yes, I can run lldb from terminal and debug.

**Reply by jsjtczh** [reply]
I found that the problem is related with third party library. I tried to write a simple program (just reading an image) using opencv, the debugger hangs at loading symbols too. It seems it cannot load symbols of a third party library.

**Reply by djairao** [reply]
Then this would be more of a problem with lldb and not with `cpptools`. Maybe there are options to skip over this 3rd party library, something similar to: https://sourceware.org/gdb/onlinedocs/gdb/Skipping-Over-Functions-and-Files.html

Don't forget to mark this thread as being answered as it is no longer a `cpptools` issue.

**Reply by jsjtczh** [reply]
I think it's a problem with cpptools, because I can use lldb in terminal to debug.

**Reply by jsjtczh** [reply]
My program and setting is similar to https://thecodinginterface.com/blog/opencv-cpp-vscode/.
